multiqc/modules/homer/tagdirectory.py

```diff
@@ -0,0 +1,568 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from HOMER tagdirectory """
+
+import logging
+import os
+import re
+from multiqc.plots import bargraph, linegraph, scatter, table
+from collections import OrderedDict
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class TagDirReportMixin():
+
+
+    def homer_tagdirectory(self):
+        """ Find HOMER tagdirectory logs and parse their data """
+
+
+
+        # Find and parse GC content:
+        for f in self.find_log_files('homer/GCcontent', filehandles=True):
+            # Get the s_name from the parent directory
+            s_name = os.path.basename(f['root'])
+            s_name = self.clean_s_name(s_name, f['root'])
+            parsed_data = self.parse_GCcontent(f)
+            if parsed_data is not None:
+                if s_name in self.tagdir_data['GCcontent']:
+                    log.debug("Duplicate GCcontent sample log found! Overwriting: {}".format(s_name))
+
+                self.add_data_source(f, s_name, section='GCcontent')
+                self.tagdir_data['GCcontent'][s_name] = parsed_data
+
+        ## get esimated genome content distribution:
+        for f in self.find_log_files('homer/genomeGCcontent', filehandles=True):
+            parsed_data = self.parse_GCcontent(f)
+            if parsed_data is not None:
+                if s_name + "_genome" in self.tagdir_data['GCcontent']:
+                    log.debug("Duplicate genome GCcontent sample log found! Overwriting: {}".format(s_name+ "_genome"))
+
+                self.add_data_source(f, s_name + "_genome", section='GCcontent')
+                self.tagdir_data['GCcontent'][s_name + "_genome"] = parsed_data
+
+
+        ## plot GC content:  
```

Comment:
To make the code a little more structured and easy to read, it would be good to group everything related to each file type into a separate function. For example:

```python
class TagDirReportMixin():
    def homer_tagdirectory(self):
        """ Find HOMER tagdirectory logs and parse their data """
        self.parse_gc_content()

    def parse_gc_content(self):
         for f in self.find_log_files('homer/GCcontent', filehandles=True):
              # parsing code etc
         # plotting code
         # code to add a section
```

..basically, to try to avoid mixing up code blocks for different parts of the report.

Reply:
done